File: Java version/app/src/main/java/it/unical/mat/parsers/pddl/PDDLParser.java
package it.unical.mat.parsers.pddl;

import it.unical.mat.parsers.pddl.pddl_parser_base.PDDLGrammarBaseVisitor;
import it.unical.mat.parsers.pddl.pddl_parser_base.PDDLGrammarLexer;
import it.unical.mat.parsers.pddl.pddl_parser_base.PDDLGrammarParser;
import java.util.ArrayList;
import org.antlr.v4.runtime.BailErrorStrategy;
import org.antlr.v4.runtime.CharStreams;
import org.antlr.v4.runtime.CommonTokenStream;
import org.antlr.v4.runtime.ConsoleErrorListener;
import org.antlr.v4.runtime.DefaultErrorStrategy;
import org.antlr.v4.runtime.RecognitionException;
import org.antlr.v4.runtime.atn.PredictionMode;

public class PDDLParser extends PDDLGrammarBaseVisitor <Void> {
    private ArrayList <String> parameters = new ArrayList <> ();
    
    private PDDLParser() {
    	
    }

    public static final PDDLParser parse(final String action) {
		final CommonTokenStream tokens = new CommonTokenStream(new PDDLGrammarLexer(CharStreams.fromString(action)));
        final PDDLGrammarParser parser = new PDDLGrammarParser(tokens);
        final PDDLParser visitor = new PDDLParser();
        
        parser.getInterpreter().setPredictionMode(PredictionMode.SLL); 
        parser.removeErrorListeners();
        parser.setErrorHandler(new BailErrorStrategy());
     
        try {
        	visitor.visit(parser.output());
        } catch (final RuntimeException exception) {
        	if(exception.getClass() == RuntimeException.class && exception.getCause() instanceof RecognitionException) {
        		tokens.seek(0);
        		parser.addErrorListener(ConsoleErrorListener.INSTANCE);
        		parser.setErrorHandler(new DefaultErrorStrategy());
        		parser.getInterpreter().setPredictionMode(PredictionMode.LL); 
        		visitor.visit(parser.output());
        	}
        }
        
        return visitor;
	}
    
    public String [] getParameters() {
        return parameters.toArray(new String [0]);
    }

    @Override
    public Void visitAtom(PDDLGrammarParser.AtomContext ctx) {
        for(int index = 1; index < ctx.IDENTIFIER().size(); index++)
            parameters.add(ctx.IDENTIFIER(index).getText());

        return null;
    }
}

File: CSharp version/it/unical/mat/parsers/pddl/PDDLParser.cs
using System;
using System.Collections.Generic;
using Antlr4.Runtime;
using Antlr4.Runtime.Atn;

namespace it.unical.mat.parsers.pddl
{
    public class PDDLParser : PDDLGrammarBaseVisitor<object>
    {
        private readonly List<string> parameters = new List<string>();

        private PDDLParser()
        {

        }

        public static PDDLParser Parse(string action)
        {
            CommonTokenStream tokens = new CommonTokenStream(new PDDLGrammarLexer(CharStreams.fromstring(action)));
            PDDLGrammarParser parser = new PDDLGrammarParser(tokens);
            PDDLParser visitor = new PDDLParser();
            parser.Interpreter.PredictionMode = PredictionMode.SLL;

            parser.RemoveErrorListeners();

            parser.ErrorHandler = new BailErrorStrategy();

            try
            {
                visitor.Visit(parser.output());
            }
            catch (SystemException exception)
            {
                if (exception.GetBaseException() is RecognitionException)
                {
                    tokens.Seek(0);
                    parser.AddErrorListener(ConsoleErrorListener<object>.Instance);

                    parser.ErrorHandler = new DefaultErrorStrategy();
                    parser.Interpreter.PredictionMode = PredictionMode.LL;

                    visitor.Visit(parser.output());
                }
            }

            return visitor;
        }

        public string[] GetParameters()
        {
            return parameters.ToArray();
        }

        public override object VisitAtom(PDDLGrammarParser.AtomContext context)
        {
            for (int index = 1; index < context.IDENTIFIER().Length; index++)
                parameters.Add(context.IDENTIFIER()[index].GetText());

            return null;
        }
    }
}